kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveDriverConfig.java

```diff
@@ -57,6 +57,18 @@ public class ResponsiveDriverConfig extends AbstractConfig {
   public static final String CLIENT_SECRET_CONFIG = "responsive.client.secret";
   private static final String CLIENT_SECRET_DOC = "The client secret for authenticated access";
 
+  // ------------------ store tuning parameters -------------------------------
+
+  // TODO: we should have another config that's applied globally, that sets a size bound on
+  //       the total amount of buffered data. That config can be used to keep a bound on
+  //       memory usage and restore times. This config sets a store-level threshold for optimal
+  //       transaction sizes.
+  public static final String STORE_FLUSH_RECORDS_THRESHOLD
```

Comment:
(for a follow-up) an additional time based one can be really useful for us when we're demo'ing/plugging in to customer staging environments that may not produce 10K records reliably - defaulting it to something like 1min is probably reasonable.